packages/emitter-framework/test/typescript/components/value-expression.test.tsx

```diff
@@ -0,0 +1,212 @@
+  // Can be replaced with with TypeKit once #6976 is implemented
+  const value = {
+    entityKind: "Value",
+    valueKind: "ArrayValue",
+    values: [$.value.createString("foo"), $.value.createNumeric(42), $.value.createBoolean(true)],
+  } as Value;
+  await testValueExpression(value, `["foo", 42, true]`);
+});
+
+it("renders scalars", async () => {
+  const program = await getProgram(`
+      namespace DemoService;
+      model DateRange {
+        @encode("rfc7231")
+        minDate: utcDateTime = utcDateTime.fromISO("2024-02-15T18:36:03Z");
+      }
+    `);
+  const [namespace] = program.resolveTypeReference("DemoService");
+  const dateRange = (namespace as Namespace).models.get("DateRange");
+  const minDate = dateRange?.properties.get("minDate")?.defaultValue;
+  assert.exists(minDate, "unable to find minDate property");
+  await testValueExpression(minDate, `"2024-02-15T18:36:03Z"`);
+});
+
+it("throws on unsupported scalar", async () => {
+  const program = await getProgram(`
+      namespace DemoService;
+
+      scalar ipv4 extends string {
+        init fromInt(value: uint32);
+      }
+
+      @example (#{ip: ipv4.fromInt(2130706433)})
+      model IpAddress {
+        ip: ipv4;
+      }
+    `);
+  const [namespace] = program.resolveTypeReference("DemoService");
+  const customScalar = (namespace as Namespace).models.get("IpAddress");
+  assert.exists(customScalar, "unable to find custom scalar");
+  const decorator = customScalar?.decorators.find((d) => d.definition?.name === "@example");
+  assert.exists(decorator?.args[0]?.value, "unable to find example decorator");
+  const value = decorator.args[0].value as Value;
+  await expect(testValueExpression(value, ``)).rejects.toThrow(
+    "Unsupported scalar constructor: fromInt",
+  );
+});
+
+it("renders empty objects", async () => {
+  // Can be replaced with with TypeKit once #6976 is implemented
+  const program = await getProgram(`
+      namespace DemoService;
+      @example(#{})
+      model ObjectValue {};
+    `);
+  const [namespace] = program.resolveTypeReference("DemoService");
+  const objectValue = (namespace as Namespace).models.get("ObjectValue");
+  const decorator = objectValue?.decorators.find((d) => d.definition?.name === "@example");
+  assert.exists(decorator?.args[0]?.value, "unable to find example decorator");
+  await testValueExpression(decorator.args[0].value as Value, `{}`);
+});
+
+it("renders objects with properties", async () => {
+  // Can be replaced with with TypeKit once #6976 is implemented
+  const program = await getProgram(`
+      namespace DemoService;
+      @example(#{a: 5, b: "foo", c: true})
+      model ObjectValue {
+        a: int32;
+        b: string;
+        c: boolean;
+      };
+    `);
+  const [namespace] = program.resolveTypeReference("DemoService");
+  const objectValue = (namespace as Namespace).models.get("ObjectValue");
+  const decorator = objectValue?.decorators.find((d) => d.definition?.name === "@example");
```

Comment:
can you make an helper in this function that just extract the value.
Check `/Users/timotheeguerin/dev/azsdk/typespec/packages/compiler/test/checker/values/utils.ts` `compilerValue` for example